x-pack/plugins/lens/public/visualizations/xy/xy_config_panel/index.tsx

```diff
@@ -72,12 +72,13 @@ const getDataBounds = function (
 ) {
   const groups: Partial<Record<string, { min: number; max: number }>> = {};
   axes.forEach((axis) => {
-    let min = Number.MAX_VALUE;
-    let max = Number.MIN_VALUE;
+    let min = Number.MAX_SAFE_INTEGER;
+    let max = -Number.MAX_SAFE_INTEGER;
     axis.series.forEach((series) => {
       activeData?.[series.layer]?.rows.forEach((row) => {
         const value = row[series.accessor];
-        if (!Number.isNaN(value)) {
+        // TODO: add tests for null value
+        if (value !== null && Number.isFinite(value)) {
```

Comment:
Here the `value` can be `null` and `Number.isNaN(null)` is actually `false` so `null` is disguised as a `number` 🥸 .

This leads to the `min` always being `0` whenever there is a `null` in the `dataset`.